lib/analytics.js

```diff
@@ -141,14 +141,21 @@ Analytics.prototype.init = Analytics.prototype.initialize = function(settings, o
   }
 
   // initialize integrations, passing ready
+  // create a list of any integrations that did not initialize - this will be passed with all events for replay support:
+  this.failedInitializations = [];
   each(function(integration) {
     if (options.initialPageview && integration.options.initialPageview === false) {
       integration.page = after(2, integration.page);
     }
 
     integration.analytics = self;
     integration.once('ready', ready);
-    integration.initialize();
+    try {
+      integration.initialize();
+    } catch (e) {
+      self.failedInitializations.push(integration.prototype.name);
+      self.log('Error initializing %s integration: %o', integration.name, e);
```

Comment:
Is it usually normal to log the whole error?

Reply:
Was going off of the same thing we do here: https://github.com/segmentio/analytics.js-integration/blob/master/lib/protos.js#L183

This is where we `try/catch` integration methods.

Reply:
This is not actually logged anywhere - it's just for debugging - so having the stack trace should be expected.

Reply:
Cool, SGTM!